From 6962dc8c6c0943ff0f3ee581f2c5c5af9f942d7c Mon Sep 17 00:00:00 2001
From: Michel JAOUEN <michel.jaouen@st.com>
Date: Tue, 18 Mar 2025 11:42:02 +0100
Subject: [PATCH 1660/1695] prod/tfm-stm32mp: add ca35_s agent

For cpufreq bl31 need access to vddcpu regu through a specific agent

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6f8ef2fefa5ba041719d3fcd77a37ebfbebe0a4d prod/tfm-stm32mp: add ca35_s agent]
Change-Id: I3624ae8445088deced9f635f7f2ca9cf2f050acf
Signed-off-by: Michel JAOUEN <michel.jaouen@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/445562
Reviewed-by: Valentin CARON <valentin.caron@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Valentin CARON <valentin.caron@foss.st.com>
---
 .../SCP-firmware/product/tfm-stm32mp2/fw/config_all.c            | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c
index f1cb0e48e..7d786a6a1 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c
@@ -504,6 +504,7 @@ static void set_scmi_comm_resources(struct scpfw_config *cfg)
 #endif
 #ifdef CFG_SCPFW_MOD_TFM_SMT
             case STM32MP25_AGENT_ID_CA35 :
+            case STM32MP25_AGENT_ID_CA35_BL31:
                 *service_data = (struct mod_scmi_service_config){
                     .transport_id = (fwk_id_t)FWK_ID_ELEMENT_INIT(FWK_MODULE_IDX_TFM_SMT, tfm_smt_index),
                     .transport_api_id = (fwk_id_t)FWK_ID_API_INIT(FWK_MODULE_IDX_TFM_SMT,
-- 
2.47.2

